iceoryx_posh/include/iceoryx_posh/roudi/introspection_types.hpp

```diff
@@ -44,14 +44,17 @@ struct MemPoolInfo
 using MemPoolInfoContainer = cxx::vector<MemPoolInfo, MAX_NUMBER_OF_MEMPOOLS>;
 
 /// @brief the topic for the mempool introspection that a user can subscribe to
-struct MemPoolIntrospectionTopic
+struct MemPoolIntrospectionInfo
 {
     uint32_t m_id;
     char m_writerGroupName[MAX_GROUP_NAME_LENGTH];
     char m_readerGroupName[MAX_GROUP_NAME_LENGTH];
     MemPoolInfoContainer m_mempoolInfo;
 };
 
+/// @brief container for MemPoolInfo structs of all available mempools.
+using MemPoolIntrospectionTopic = cxx::vector<MemPoolIntrospectionInfo, MAX_SHM_SEGMENTS>;
```

Comment:
this must be `MAX_SHM_SEGMENTS + 1` since there is the additional mempool for the introspection samples